spark/dl/src/test/scala/com/intel/analytics/bigdl/torch/VolumetricAveragePoolingSpec.scala

```diff
@@ -0,0 +1,335 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.torch
+
+import com.intel.analytics.bigdl._
+import com.intel.analytics.bigdl.nn.{GradientChecker, VolumetricAveragePooling}
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.utils.RandomGenerator._
+
+import scala.util.Random
+
+@com.intel.analytics.bigdl.tags.Serial
+class VolumetricAveragePoolingSpec extends TorchSpec {
+
+  "VolumetricAveragePooling Forward dim 4 Double" should "work properly" in {
+    torchCheck()
+    val seed = 100
+    RNG.setSeed(seed)
+    val from = RNG.uniform(2, 4).toInt
+    val to = RNG.uniform(1, 4).toInt
+    val kt = RNG.uniform(2, 4).toInt
+    val ki = RNG.uniform(2, 4).toInt
+    val kj = RNG.uniform(2, 4).toInt
+    val st = RNG.uniform(1, 3).toInt
+    val si = RNG.uniform(1, 3).toInt
+    val sj = RNG.uniform(1, 3).toInt
+    val padT = Math.min(RNG.uniform(0, 2).toInt, math.floor(kt / 2).toInt)
+    val padW = Math.min(RNG.uniform(0, 2).toInt, math.floor(kt / 2).toInt)
+    val padH = Math.min(RNG.uniform(0, 2).toInt, math.floor(kt / 2).toInt)
+    val outt = RNG.uniform(5, 7).toInt
+    val outi = RNG.uniform(5, 7).toInt
+    val outj = RNG.uniform(5, 7).toInt
+    val batch = RNG.uniform(2, 7).toInt
+    val int = (outt - 1) * st + kt - padT * 2
+    val ini = (outi - 1) * si + ki - padW * 2
+    val inj = (outj - 1) * sj + kj - padH * 2
+    val layer = VolumetricAveragePooling[Double](kt, ki, kj, st, si, sj,
+      padT, padW, padH)
+
+    val input = Tensor[Double](from, int, ini, inj).apply1(e => Random.nextDouble())
+    val output = layer.updateOutput(input)
+
+    val code = "torch.manualSeed(" + seed + ")\n" +
+      s"layer = nn.VolumetricAveragePooling($kt, $ki, $kj, $st, $si, $sj, $padT," +
+      s" $padW, $padH)\n" +
+      "output = layer:forward(input)"
+
+    val (luaTime, torchResult) = TH.run(code, Map("input" -> input),
+      Array("output"))
+
+    val luaOutput = torchResult("output").asInstanceOf[Tensor[Double]]
+
+    output.size() should be(luaOutput.size())
+    output should be(luaOutput)
+  }
+
+  "VolumetricAveragePooling Forward dim 5 Double" should "work properly" in {
```

Comment:
Somehow we can extract this testing code and remove duplication to make it shorter, but I guess we can do this together with the other Scala test in the next PR.

Reply:
Yes, if we want to shorten code for scala test, we can extract a function and use it for other tests as well.